modules/transport-netty4/src/main/java/org/elasticsearch/transport/NettyAllocator.java

```diff
@@ -79,12 +110,30 @@ public static ByteBufAllocator getAllocator() {
         }
     }
 
-    private static boolean useUnpooled() {
+    private static boolean useUnpooled(long heapSizeInBytes, long g1RegionSize) {
+        if (userForcedUnpooled()) {
+            return true;
+        } else {
+            boolean heapIsOneGBOrLess = heapSizeInBytes <= 1 << 30;
+            boolean g1gcRegionIsLessThan1MB = g1RegionSize < 1 << 20;
+            boolean unknownRegionSize = g1RegionSize != -1;
+            return heapIsOneGBOrLess || (g1gcRegionIsLessThan1MB && unknownRegionSize);
```

Comment:
I think this should be:

```suggestion
            return heapIsOneGBOrLess || (g1gcRegionIsLessThan1MB && unknownRegionSize == false);
```